happi/utils.py

```diff
@@ -118,3 +119,21 @@ def warn_and_call(*args, **kwargs):
         return warn_and_call
 
     return wrapper
+
+
+def build_abs_path(basedir: str, path: str) -> str:
+    """
+    Builds an abs path starting at basedir if path is not already absolute.
+    If path is absolute already, this function returns path without modification.
+
+    Parameters
+    ----------
+    basedir : str
+        If path is not absolute already, build an abspath
+        with path starting here.
+    path : str
+        The path to convert to absolute.
+    """
+    if not os.path.isabs(path):
+        return os.path.abspath(basedir + '/' + path)
```

Comment:
suggest join again for platform compatibility
```suggestion
        return os.path.abspath(os.path.join(basedir, path))
```